src/main/webapp/app/shared/components/eventTable/EventTableRow.tsx

```diff
@@ -1,51 +1,59 @@
 import React from 'react';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 import { Button } from 'reactstrap';
-import { IEventCrew } from 'app/shared/model/event-crew.model';
-import { IEventAttendee } from 'app/shared/model/event-attendee.model';
-import { translate } from 'react-jhipster';
+import { IColumns } from '../../model/columns.model';
 
 export interface IEventTableRowProps {
-  user: IEventCrew | IEventAttendee;
+  hasNumbering?: boolean;
   index: number;
-  openModal: Function;
+  columns: IColumns[];
+  data: { [key: string]: any };
+  whatsappKey?: string;
+  action?: (id: number) => void;
 }
 
 export class EventTableRow extends React.Component<IEventTableRowProps> {
-  state = { user: this.props.user };
-  onToggleModal = () => this.props.openModal(this.props.user.id);
-
-  contactUser = () => {
-    window.open(`https://wa.me/+60${this.props.user.contactNumber}`, '_blank');
-  };
+  constructor(props: IEventTableRowProps) {
+    super(props);
+    this.action = this.action.bind(this);
+    this.contactUser = this.contactUser.bind(this);
+  }
 
-  render() {
-    const { user, index } = this.props;
+  contactUser(): void {
+    if (this.props.whatsappKey) {
+      window.open(`https://wa.me/+60${this.props.data[this.props.whatsappKey]}`, '_blank');
+    }
+  }
 
-    let thirdColumn;
-    if ('role' in user) {
-      thirdColumn = translate(`clubmanagementApp.EventCrewRole.${user.role}`);
-    } else if ('year' in user) {
-      // @ts-ignore
-      thirdColumn = user.year;
+  action(): void {
+    if (this.props.action) {
+      this.props.action(this.props.data.id);
     }
+  }
+
+  render() {
+    const { data, columns, hasNumbering, whatsappKey, index, action } = this.props;
 
     return (
       <tr>
-        <td scope="row">{index + 1}</td>
-        <td>{user.userName}</td>
-        <td>{thirdColumn}</td>
-        <td>{'provideTransport' in user ? <FontAwesomeIcon icon="car" /> : null}</td>
-        <td>
-          <Button color="Link" className="icon-btn" onClick={this.contactUser}>
-            <FontAwesomeIcon icon={['fab', 'whatsapp-square']} color="#25D366" size="lg" />
-          </Button>
-        </td>
-        <td>
-          <Button onClick={this.onToggleModal} color="Link" className="icon-btn">
-            <FontAwesomeIcon icon="ellipsis-v" color="#07ade1" />
-          </Button>
-        </td>
+        {hasNumbering && <td scope="row">{index + 1}</td>}
+        {columns.map(column => (
+          <td key={data[column.key]}>{column.replaceValue && data[column.key] ? column.replaceValue : data[column.key]}</td>
```

Comment:
Because the replaceValue will be used to replace the data value if the data value exists. The use case is as the car icon.
It's not meant to use as default value but to replace the value if the value exist.

Reply:
I got it
But look a bit confuse at the beginning
Nvm proceed first